tests/integration/contracts.test.js

```diff
@@ -245,5 +251,158 @@ describe('Contracts', () => {
+          extraDataContractInstance
+        )
+
+        // raise dispute party A
+        const raiseDisputeByPartyATxObj = await ArbitrableTransactionInstance.payArbitrationFeeByPartyA(
+          partyA,
+          arbitrationCost -
+            web3.fromWei(partyAFeeContractInstance, 'ether').toNumber()
+        )
+        expect(raiseDisputeByPartyATxObj.tx).toEqual(
+          expect.stringMatching(/^0x[a-f0-9]{64}$/)
+        ) // tx hash
+
+        // raise dispute party B
+        const raiseDisputeByPartyBTxObj = await ArbitrableTransactionInstance.payArbitrationFeeByPartyB(
+          partyB,
+          arbitrationCost -
+            web3.fromWei(partyBFeeContractInstance, 'ether').toNumber()
+        )
+        expect(raiseDisputeByPartyBTxObj.tx).toEqual(
+          expect.stringMatching(/^0x[a-f0-9]{64}$/)
+        ) // tx hash
+
+        // ****** Juror side (pass period) ****** //
+
+        let newPeriod
+        // pass state so jurors are selected
+        for (let i = 1; i < 3; i++) {
+          // NOTE we need to make another block before we can generate the random number. Should not be an issue on main nets where avg block time < period length
+          if (i === 2)
+            web3.eth.sendTransaction({
+              from: partyA,
+              to: partyB,
+              value: 10000,
+              data: '0x'
+            })
+          await delaySecond()
+          await KlerosPOCInstance.passPeriod()
+
+          newPeriod = await KlerosPOCInstance.getPeriod()
+          expect(newPeriod).toEqual(i)
+        }
+
+        let drawA = []
+        let drawB = []
+        for (let i = 1; i <= 3; i++) {
+          if (
+            await KlerosPOCInstance.isJurorDrawnForDispute(0, i, jurorContract1)
+          ) {
+            drawA.push(i)
+          } else {
+            drawB.push(i)
+          }
+        }
+
+        const rulingJuror1 = 2 // vote for partyB
+        await KlerosPOCInstance.submitVotes(
+          0,
+          rulingJuror1,
+          drawA,
+          jurorContract1
+        )
+
+        const rulingJuror2 = 2 // vote for partyB
+        await KlerosPOCInstance.submitVotes(
+          0,
+          rulingJuror2,
+          drawB,
+          jurorContract2
+        )
+
+        await delaySecond()
+        await KlerosPOCInstance.passPeriod(other)
+
+        const currentRuling = await klerosPOCInstance.currentRuling(0)
+        expect(`${currentRuling}`).toEqual('2') // partyB wins
+
+        const appealCost = await KlerosPOCInstance.getAppealCost(
+          0,
+          extraDataContractInstance
+        )
+
+        // raise appeal party A
+        const raiseAppealByPartyATxObj = await ArbitrableTransactionInstance.appeal(
+          partyA,
+          extraDataContractInstance,
+          appealCost
+        )
+        expect(raiseAppealByPartyATxObj.tx).toEqual(
+          expect.stringMatching(/^0x[a-f0-9]{64}$/)
+        ) // tx hash
+      },
```

Comment:
At the end it would be good to grab the dispute from arbitrator and make sure that dispute.appeals has incremented by 1.